### PR TITLE
add unit test for database discovery namespace module

### LINKED ISSUE #14496: The db-discovery namespace need add unit test.
Hi community,

The `shardingsphere-db-discovery-spring-namespace` module need add unit test.  You can refer `shardingsphere-readwrite-splitting-spring-namespace` module to finish it.

### PR DESCRIPTION
**Pull request by LeeGuoPing**
Fixes #14496.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+@ContextConfiguration(locations = "classpath:META-INF/spring/database-discovery-application-context.xml")
+public final class DatabaseDiscoverySpringNamespaceTest extends AbstractJUnit4SpringContextTests {
+    
+    @Resource
+    private DatabaseDiscoveryType mGRDatabaseDiscoveryType;
```
Comment (zjcnb) [inline shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java:41]: @LeeGuoPing  Hi, thank you for your pull request. `mGRDatabaseDiscoveryType` maybe change to `MGRDatabaseDiscoveryType`is better.

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    
+    @Resource
+    private AlgorithmProvidedDatabaseDiscoveryRuleConfiguration defaultRule;
+    
+    @Test
+    public void assertMgrDatabaseDiscoveryType() {
```
Comment (zjcnb) [inline shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java:47]: chage to` assertMGRDatabaseDiscoveryType` is better.

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java (hunk 1 above)
Comment (LeeGuoPing) [reply shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java:41]: Why is this word capitalized?

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java (hunk 2 above)
Comment (LeeGuoPing) [reply shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java:47]: fine

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java (hunk 1 above)
Comment (terrymanu) [reply shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java:41]: `mgrDatabaseDiscoveryType` is better

Comment (LeeGuoPing) [reply shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-spring/shardingsphere-db-discovery-spring-namespace/src/test/java/org/apache/shardingsphere/db/discovery/spring/namespace/DatabaseDiscoverySpringNamespaceTest.java]: ok @terrymanu

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**